### PR TITLE
Add Eoip packege

### PR DESCRIPTION
**Pull request by bogdik**
Maintainer: Bogdik <bogdikxxx@gmail.com>
Compile tested: zbt wr8305rt, dlink dir-300, dlink dir-615
Run tested: zbt wr8305rt mips, OpenWrt last snapshot, tests done

Description: Add Eoip package movement from https://github.com/openwrt/openwrt/pull/3721

### CODE REVIEW

File: net/eoip/patches/100-add_CR_to_pidfile.patch
Comment (musashino205) [inline net/eoip/patches/100-add_CR_to_pidfile.patch:8]: ```\n``` is not a CR, and why do you need this change?

Comment (bogdik) [reply net/eoip/patches/100-add_CR_to_pidfile.patch:8]: I dont know,but this change on main repository used https://github.com/agustim/openwrt-linux-eoip/commit/1319744c84ea7ccc7ed46a1a8203cc4fe65c22d4

**Comment by bobafetthotmail** [comment]
you need to add PKG_MAINTAINER:= My Name <myemail@email.com> to the makefile as other packages in the package feed. These packages aren't maintained by OpenWrt developers but by community.

File: net/eoip/patches/100-add_CR_to_pidfile.patch
Comment (bogdik) [reply net/eoip/patches/100-add_CR_to_pidfile.patch:8]: > `\n` is not a CR, and why do you need this change?

you think you need to remove the patch?

Comment (musashino205) [reply net/eoip/patches/100-add_CR_to_pidfile.patch:8]: To be honest, I can't judge.
I'm not using eoip so I'm not sure it will work without the patch, and eoip's official repositories do have this change, but there's no detailed explanation...
https://code.google.com/archive/p/linux-eoip/source/default/commits

BTW, I recommend changing ```CR``` to ```LF``` in the patch name.

Comment (bogdik) [reply net/eoip/patches/100-add_CR_to_pidfile.patch]: done

File: net/eoip/Makefile
Comment (hnyman) [inline net/eoip/Makefile:2]: Copyright years look strange. Is this 8 years old? 
Is the package being imported from the oldpackages repo? If so, that should be mentioned.

Is the upstream still alive?
Is the actual underlying package still being developed?  (or is this old abandonware?)

File: net/eoip/Makefile
Comment (hnyman) [inline net/eoip/Makefile:29]: As you only have a normal Package/eoip package, there is no need to a separate rather empty Package/eoip/Default definition. You should combine its contents here.

File: net/eoip/Makefile
Comment (hnyman) [inline net/eoip/Makefile:16]: Shouldn't that be https ?

File: net/eoip/Makefile
Comment (hnyman) [inline net/eoip/Makefile:25]: Using that link, the upstream license is said to be "License: GNU GPL v3"

You should correct the PKG_LICENSE license definition in Makefile

File: net/eoip/Makefile
Comment (hnyman) [inline net/eoip/Makefile:11]: If you later set the download for version 0.6.1, you should set also PKG_VERSION here to 0.6.1

Or, if you are piggybacking with an upstream source archive of version 0.5, you should still call this as version 0.5 instead of calling this as "0.6.1" and then providing a tar.gz of version 0.5 copied from upper upstream...

That looks strange in any case. It pretty much looks like your download site & "release 0.6.1" is just a storage of an old tar.gz downloaded from upstream. (instead of you really hosting the underlying source code as plain source).

File: net/eoip/files/eoip.init
Comment (hnyman) [inline net/eoip/files/eoip.init:36]: Is the mixing of leading spaces and tabs intentional?

File: net/eoip/patches/100-add_LF_to_pidfile.patch
Comment (hnyman) [inline net/eoip/patches/100-add_LF_to_pidfile.patch:1]: Is the source file to be patched really named `eoip.c.orig`  ?
I would assume that this should be `eoip.c` 

(Looking at your upper upstream source code archive, there is no file  `eoip.c.orig` )

File: net/eoip/Makefile
Comment (hnyman) [inline net/eoip/Makefile:17]: From automatic CI testing:
* WARNING: Use of PKG_MD5SUM is deprecated, switch to PKG_HASH
* WARNING: PKG_MD5SUM uses deprecated hash, set to 22f6f3864665adef26c7fbd57543a396108ba2dff1282af8143f18bc2a9912f8

-->
PKG_HASH:=22f6f3864665adef26c7fbd57543a396108ba2dff1282af8143f18bc2a9912f8

**Review by hnyman — CHANGES_REQUESTED** [review]
The package needs work, and I am not quite sure if the actual upstream package is still alive.

File: net/eoip/Makefile
Comment (bogdik) [reply net/eoip/Makefile:2]: no, it's just a fork that I was working on https://github.com/bogdik/openwrt-linux-eoip

File: net/eoip/patches/100-add_LF_to_pidfile.patch
Comment (bogdik) [reply net/eoip/patches/100-add_LF_to_pidfile.patch:1]: https://github.com/agustim/openwrt-linux-eoip/commit/2e433dd9156c672d7128f95ce624781485cb44d7

File: net/eoip/Makefile
Comment (bogdik) [reply net/eoip/Makefile:11]: > One more comment: there needs to be just one commit. (Currently you have 5 commits that modify only 4 files...)

This is the first time I come across pull request , and by package openwrt requirements, I may close this one and create a new one, if it need

**Review by bogdik — COMMENTED** [review]
I sort of fixed all the questions, if you need to create a new request, I'll do

File: net/eoip/Makefile
Comment (bobafetthotmail) [reply net/eoip/Makefile]: That's a SHA-256 right?

Comment (hnyman) [reply net/eoip/Makefile]: Yep.
PKG_MD5SUM has been deprecated years ago

**Comment by hnyman** [comment]
> I sort of fixed all the questions

Not quite:

* the patch is still wrong.  There is no eiop.c.orig in the source tar.gz, even if you have seen that erroneous patch somewhere upstream.

* You need to squash the 7 commits into one, and update the PR by "git push --force" . It would be easier if you would have followed the advice and used a separate branch for the PR (instead of using local master). But the advice about git rebase still mainly works for you: 
  https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#advice-on-pull-requests

* PKG_LICENSE line is still wrong. It should be the correct SPDX code from the end of the CONTRIBUTING.md page.

* The line `TITLE+= (command line)` looks wrong, as you already have TITLE in the same section.

In general, I am still worried that this is old abandonware.

**Comment by bobafetthotmail** [comment]
What is the performance?
There is a kernel patch to add support for Eoip in Linux kernel that is maintained/developed here https://github.com/bbonev/eoip and they claim that their project solves a "performance problem" that userspace applications have.

**Comment by bogdik** [comment]
> What is the performance?
> There is a kernel patch to add support for Eoip in Linux kernel that is maintained/developed here https://github.com/bbonev/eoip and they claim that their project solves a "performance problem" that userspace applications have.
> And it also has been maintained since 2013 so it's probably not abandonware.

I don't have gigabit devices, but those that I have at my disposal have no performance problems, there is also a ready-made management system luci, the kernel patch needs to be ported / integrated, but I think people should have a choice of what to use

**Comment by bogdik** [comment]
> In general, I am still worried that this is old abandonware.

I am making efforts and I do not understand whether it makes sense if you are not determined to add this software, I get a lot of requests why it is not possible to select it from the main repository, I tried to do it, but apparently it is useless

**Comment by bobafetthotmail** [comment]
> > In general, I am still worried that this is old abandonware.
> 
> I am making efforts and I do not understand whether it makes sense if you are not determined to add this software, I get a lot of requests why it is not possible to select it from the main repository, I tried to do it, but apparently it is useless

even if he disagrees, others can merge the package in this repository.

You still need to do what he said in the post above to clean up this contribution patches because these are the contribution rules. See here for the full rules https://openwrt.org/submitting-patches

**Comment by bogdik** [comment]
> You still need to do what he said in the post above to clean up this contribution patches as there are contribution rules. See here for the full rules https://openwrt.org/submitting-patches

I done all request, if no need any action, i do squash commits

**Comment by antonpopov** [comment]
> The package needs work, and I am not quite sure if the actual upstream package is still alive.
> 
> One more comment: there needs to be just one commit. (Currently you have 5 commits that modify only 4 files...)

@bogdik have you managed to deal with the suggested changes? I'm also interested in EoIP under openwrt. Your work will be much appreciated.

**Comment by hnyman** [comment]
You still need to squash the commits into one, and place a proper commit message including a signed-off-by line into it.

(also the Maintainer field in Makefile should have your real name, similarly as the signed-off-by line in the commit.)

**Comment by bogdik** [comment]
Uhh it was hard

**Comment by antonpopov** [comment]
@bogdik  are you sure everything went good after merging the pull requests? Some of the changes are now applied.
 image

**Comment by bobafetthotmail** [comment]
> Uhh it was hard

Github's interface is limited, these operations are easier from command line, see the instructions here https://openwrt.org/docs/guide-developer/working-with-github-pr

**Comment by bogdik** [comment]
> > Uhh it was hard
> 
> Github's interface is not very good, these operations are easier from command line, see the instructions here https://openwrt.org/docs/guide-developer/working-with-github-pr

yes but your request not compiled this first (you test it?), and second read our discussion thread, for tabs fix thanks

**Comment by bogdik** [comment]
> Github's interface is not very good, these operations are easier from command line, see the instructions here

I would say that without the command line the issues could not be resolved, well, I kind of did everything as said, here my powers end

File: net/eoip/Makefile
Comment (BKPepe) [inline net/eoip/Makefile:19]:
```suggestion
PKG_NAME:=eoip
PKG_VERSION:=0.5
PKG_RELEASE:=1

PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://github.com/bogdik/openwrt-linux-eoip/releases/download/0.5/
PKG_HASH:=22f6f3864665adef26c7fbd57543a396108ba2dff1282af8143f18bc2a9912f8

PKG_MAINTAINER:=bogdik <bogdikxxx@mail.ru>
PKG_LICENSE:=GPL-3.0-or-later
PKG_INSTALL:=1
```

File: net/eoip/Makefile
Comment (BKPepe) [inline net/eoip/Makefile:25]:
```suggestion
  URL:=https://code.google.com/p/linux-eoip
```

File: net/eoip/Makefile
Comment (BKPepe) [inline net/eoip/Makefile:41]: This is not needed, right?

File: net/eoip/Makefile
Comment (BKPepe) [inline net/eoip/Makefile:6]: Omit it or add your own copyright.

See https://github.com/openwrt/packages/blob/master/CONTRIBUTING.md#makefile-contents-should-contain

**Review by BKPepe — CHANGES_REQUESTED** [review]
I have two things:

- Can you please add in your commit description, why do we need this package in OpenWrt and what it does?

- Why we are using your fork? Did you try to send your changes (differences) to repository, which you forked - https://github.com/agustim/openwrt-linux-eoip ? I see that owner of that repository responds to it.

File: net/eoip/Makefile
Comment (BKPepe) [reply net/eoip/Makefile:19]: As mentioned above, you should use in `PKG_MAINTAINER` your real name.

**Comment by bogdik** [comment]
> I have two things:
> 
>     * Can you please add in your commit description, why do we need this package in OpenWrt and what it does?
> 
>     * Why we are using your fork? Did you try to send your changes (differences) to repository, which you forked - https://github.com/agustim/openwrt-linux-eoip ? I see that owner of that repository responds to it.

Perhaps I'll close this request, it's impossible! already some kind of mockery, who needs to collect themselves

**Comment by bobafetthotmail** [comment]
> > I have two things:
> > ```
> > * Can you please add in your commit description, why do we need this package in OpenWrt and what it does?
> > ```
> 
> Perhaps I'll close this request, it's impossible! already some kind of mockery, who needs to collect themselves
> 

This was in the contribution rules. Other packages that were added have a bit of description in the commit message https://github.com/openwrt/packages/commit/509b9d52a08fa7df457180180d2934d4ba6f9a41

**Comment by hnyman** [comment]
Long process, but now it looks good enough.
I will merge it.

File: net/eoip/Makefile
Comment (BKPepe) [reply net/eoip/Makefile]: @hnyman This was not applied. :(

Comment (hnyman) [reply net/eoip/Makefile]: > This was not applied. :(

What was not applied?

`PKG_MAINTAINER:=Bogdan Shatik <bogdikxxx@mail.ru>`

Comment (BKPepe) [reply net/eoip/Makefile]: Yes, he corrected the maintainer, but he didn't apply my suggestion to increase the stylistic of Makefile but never mind. And would be great if he can send his changes to upstream repository, so we know that the upstream repository is not dead.

**Comment by BKPepe** [comment]
Thanks for adding description.

**Comment by antonpopov** [comment]
@bogdik congrats!

**Pull request merged**
